core/templates/pages/android-page/android-page.component.spec.ts

```diff
@@ -159,41 +159,60 @@ describe('Android page', () => {
     expect(component.validateEmailAddress()).toBeTrue();
   });
 
-  it('should catch error when adding user to android mailing list',
+  it('should add user to android mailing list and return status',
     fakeAsync(() => {
       spyOn(alertsService, 'addInfoMessage');
       component.ngOnInit();
       tick();
       component.emailAddress = 'validEmail@example.com';
       component.name = 'validName';
       spyOn(androidUpdatesBackendApiService, 'subscribeUserToAndroidList')
-        .and.returnValue(Promise.reject({
-          error: { error: 'Internal Server Error.'}
-        }));
+        .and.returnValue(Promise.resolve(true));
 
       component.subscribeToAndroidList();
 
       flushMicrotasks();
 
-      expect(alertsService.addInfoMessage).not.toHaveBeenCalled();
+      expect(alertsService.addInfoMessage).toHaveBeenCalledWith(
+        'Done!', 1000);
     }));
 
-  it('should add user to android mailing list and return status',
+  it('should fail to add user to android mailing list and return status',
     fakeAsync(() => {
       spyOn(alertsService, 'addInfoMessage');
       component.ngOnInit();
       tick();
       component.emailAddress = 'validEmail@example.com';
       component.name = 'validName';
       spyOn(androidUpdatesBackendApiService, 'subscribeUserToAndroidList')
-        .and.returnValue(Promise.resolve());
+        .and.returnValue(Promise.resolve(false));
 
       component.subscribeToAndroidList();
 
       flushMicrotasks();
 
       expect(alertsService.addInfoMessage).toHaveBeenCalledWith(
-        'Done!', 1000);
+        'Sorry, some unexpected occurred. Please email admin@oppia.org to ' +
```

Comment:
Sorry, an unexpected error occurred. Please email ... to be added to the mailing list.

Reply:
Done